aten/src/ATen/native/quantized/cpu/qnnpack/CMakeLists.txt

```diff
@@ -0,0 +1,684 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5 FATAL_ERROR)
+
+INCLUDE(GNUInstallDirs)
+
+# ---[ Project and semantic versioning.
+PROJECT(PYTORCH_QNNPACK C CXX ASM)
```

Comment:
you can also just drop it as we'd be compiling it inline (but not sure)

Reply:
Yes, that is true.  My goal was to minimize modifications to make sure I am not breaking anything in the process inadvertently since different flags may be used between PyTorch and QNNPACK for compilation and linking, but I can definitely try that if that's what you prefer.

Reply:
> Have we tried to compile it with C++? It might be easier that way as we can use namespaces.

I actually asked the same question about namespace offline :) I don't know how much work to recompile with c++ though. In worst case we can change it back to namespace when things are more clear?

Reply:
OK so I tried compiling with C++, but it's a good deal of work.  The compiler is complaining about jumps (use of goto to be exact) bypassing variable initializations, and use of 'restrict' (as it's not a C++ keyword.)   All of this can be worked around (using __restrict for instance instead of restrict and initializing all these variables to null / zero prior at them being jumped over), but it's a good deal of work.  Making these changes is actually one thing and not terribly difficult, but making sure they do not introduce any hidden bugs is a whole another can of worms that I would rather not open at this point.

https://gist.github.com/AshkanAliabadi/d7e78ce6c0ea6e9b092d71a9bb96b9de

Reply:
It would be nice to compile with C++ for new files. I am planning on adding new APIs to make QNNPACK more functional and it would be a bit clunky if they are written in C as well. That can be part of a different PR though.

Reply:
We shouldn't have any problems adding new C++ files to QNNPack.  QNNPack already has C++ files in the form of tests and benchmarks.  It is just that modifying the C files to make them C++-conformant is more work that we can take on right now.